core/server/master/src/main/java/alluxio/master/block/BlockMasterWorkerServiceHandler.java

```diff
@@ -61,11 +67,13 @@ public BlockMasterWorkerServiceHandler(BlockMaster blockMaster) {
   @Override
   public void blockHeartbeat(BlockHeartbeatPRequest request,
       StreamObserver<BlockHeartbeatPResponse> responseObserver) {
-    if (LOG.isDebugEnabled()) {
-      LOG.debug("Block heartbeat request is {} bytes, {} added blocks and {} removed blocks",
-              request.getSerializedSize(),
-              request.getAddedBlocksCount(),
-              request.getRemovedBlockIdsCount());
+    if (request.getSerializedSize() > RPC_REQUEST_SIZE_WARNING_THRESHOLD) {
+      LOG.debug("blockHeartbeat request is {} bytes, {} added blocks, "
+                    + "{} removed blocks, {} metrics",
+          request.getSerializedSize(),
+          request.getAddedBlocksCount(),
+          request.getRemovedBlockIdsCount(),
+          request.getOptions().getMetricsCount());
     }
```

Comment:
can we refactor this into a Util method? e.g.

```
public static void logRequestIfTooLarge(GeneratedMessageV3 request, String message, Object... args) {
   if (request.getSerializedSize() > RPC_REQUEST_SIZE_WARNING_THRESHOLD) {
      LOG.warn(messages, args);
   }
}
```

and, why it is LOG.warn for response but LOG.debug for request?

Reply:
why it is LOG.warn for response but LOG.debug for request? 
 - That's my typo :(

Reply:
I feel like `SamplingLogger` is a better fit for these logs in general.